src/WebJobs.Extensions.EasyTables/Bindings/EasyTableAttributeBindingProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.WindowsAzure.MobileServices;
+using WebJobs.Mobile.EasyTables;
+
+namespace WebJobs.Extensions.EasyTables
+{
+    internal class EasyTableAttributeBindingProvider : IBindingProvider
+    {
+        private EasyTableConfiguration _easyTableConfig;
+        private JobHostConfiguration _jobHostConfig;
+        private INameResolver _nameResolver;
+
+        public EasyTableAttributeBindingProvider(JobHostConfiguration config, EasyTableConfiguration easyTableConfig, INameResolver nameResolver)
+        {
+            _jobHostConfig = config;
+            _easyTableConfig = easyTableConfig;
+            _nameResolver = nameResolver;
+        }
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            ParameterInfo parameter = context.Parameter;
+            EasyTableAttribute attribute = parameter.GetCustomAttribute<EasyTableAttribute>(inherit: false);
+            if (attribute == null)
+            {
+                return Task.FromResult<IBinding>(null);
+            }
+
+            if (_easyTableConfig.MobileAppUri == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The Easy Tables Uri must be set either via a '{0}' app setting or directly in code via EasyTableConfiguration.MobileAppUri.",
+                    EasyTableConfiguration.AzureWebJobsMobileAppUriName));
+            }
+
+            EasyTableContext easyTableContext = CreateContext(_easyTableConfig, attribute, _nameResolver);
+
+            IBindingProvider compositeProvider = new CompositeBindingProvider(new IBindingProvider[]
```

Comment:
TableAttributeBindingProvider provides a lot of different bindings. I based some of this on how that worked.

I want [EasyTable] to act similarly -- one attribute that serves different purposes based on the parameter type.

Is there a different approach to take?

---
In reply to: [54809415](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/45#discussion_r54809415) [](ancestors = 54809415)